practice/flutter/flutter/logic.py

```
@@ -0,0 +1,245 @@
+
+    :raise KeyError
+    :return models.User
+    >>> user = User.objects.create_user('foo', password='letmein')
+    >>> get_user_by_username('foo').username == 'foo'
+    True
+    >>> get_user_by_username('FOO', case_sensitive=False).username == 'foo'
+    True
+    >>> get_user_by_username('FOO')
+    Traceback (most recent call last):
+    ...
+    LookupError: user_not_found|FOO
+    """
+    err_str = 'user_not_found|' + username
+    try:
+        if case_sensitive:
+            user = models.User.objects.get(username=username)
+        else:
+            user = models.User.objects.get(username__iexact=username)
+
+        get_fluttuser_from_stduser(user)
+    except models.User.DoesNotExist:
+        raise LookupError(err_str)
+
+    return user
+
+
+def get_fluttuser_from_stduser(std_user: User):
+    """ Ensure that FluttUser is created """
+    try:
+        fu = std_user.flutteruser
+    except models.FlutterUser.DoesNotExist:
+        fu = models.FlutterUser(user=std_user)
+        fu.save()
+
+    return fu
+
+
+def preprocess_flutt(text: str):
+    """ Pre-process text minimally before saving
+
+    :param text
+    :return: str - filtered text
+    >>> preprocess_flutt(' This  is <em>not</em> the Markdown we are looking for')
+    'This is not the Markdown we are looking for'
+    """
+    processed = re.sub(r'\s+', ' ', text.strip())  # type: str
+    processed = re.sub(r'<[^>]+>', '', processed)
+    return processed
+
+
+def validate_flutt(text: any) -> any:
+    """ Validate text for saving
+
+    :raise: ValueError if invalid
+    :returns: processed text if valid, False if invalid, None if None passed
+    >>> validate_flutt(' This flutt has issues, but it  is valid.  ')
+    'This flutt has issues, but it is valid.'
+    >>> validate_flutt('<script src="http://evil.website/">')
+    Traceback (most recent call last):
+    ...
+    ValueError: invalid_flutt
+    """
+    if text is None:
+        flutt = text
```

Then my suggestion is to remove it to prevent it from swallowing any errors.